core/src/test/scala/unit/kafka/log/remote/RemoteIndexCacheTest.scala

```diff
@@ -132,7 +133,7 @@ class RemoteIndexCacheTest {
     // this call should have invoked fetchOffsetIndex, fetchTimestampIndex once
     val resultPosition = cache.lookupOffset(rlsMetadata, offsetPosition1.offset)
     assertEquals(offsetPosition1.position, resultPosition)
-    verifyFetchIndexInvocation(count = 1, Seq(IndexType.OFFSET, IndexType.TIMESTAMP))
+    verifyFetchIndexInvocation(count = 1, Seq(IndexType.OFFSET))
```

Comment:
Checking fetchIndex invocation for Offset index type only as fetchIndex mock is creating all index files during the first invocation so for remaining index types there will be no fetchIndex Invocation.

Reply:
In the fetchIndex mock (in setup()), irrespective of the IndexType passed in the parameter, we are creating all the index files in the cacheDir and due to that, during cacheMiss, fetchIndex is getting invoked during OFFSET index fetch only and for other two index types, fetchIndex is not getting called as the index is getting fetched from the disk only.

These test cases were passing earlier, as before we were creating those index files in tmpDir, not in the actual dir used for RemoteIndexCache, and after I fixed that issue, test cases started to fail, so I had to make this change and similar changes in other test cases as well.

Reply:
@iit2009060 Got it. I have reverted these changes.